custom_components/monitor_docker/sensor.py

```diff
@@ -226,16 +241,27 @@ def __init__(
         self._attributes: dict[str, Any] = {}
         self._removed = False
 
+        api_info = self._api.get_info()
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._api.get_url())},
+            name=self._instance,
+            model=api_info.get(ATTR_VERSION_OS),
+            manufacturer="Docker",
+            model_id=api_info.get(ATTR_VERSION_KERNEL),
+            sw_version=api_info.get(DOCKER_INFO_VERSION),
+            entry_type=DeviceEntryType.SERVICE,
+        )
+
         _LOGGER.info(
             "[%s]: Initializing Docker sensor '%s'",
             self._instance,
             self.entity_description.name,
         )
 
-    @property
-    def entity_id(self) -> str:
-        """Return the entity id of the sensor."""
-        return self._entity_id
+    # @property
+    # def entity_id(self) -> str:
+    #     """Return the entity id of the sensor."""
+    #     return self._entity_id
```

Comment:
With removal of this and the change to set _attr_unique_id instead of _entity_id I would fear that the config-parameter `rename_entity` becomes obsolete, but was necessary since the device_registry seem to need controll over the entity naming. Could be solved by setting self_entity_id (instead of self._entity_id) and hence making it a public read/write property.